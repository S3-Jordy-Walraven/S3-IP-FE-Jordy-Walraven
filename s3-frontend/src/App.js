import "./App.css";
import NavigationBar from "./components/layout/NavigationBar";
import "../node_modules/bootstrap/dist/css/bootstrap.min.css";
import "./ColorStyle.css";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import Homepage from "./pages/GeneralPages/Homepage";
import LoginPage from "./pages/GeneralPages/LoginPage";
import { useState, useEffect } from "react";
import { userContext } from "./context/userContext";
import EffectUploadPage from "./pages/EffectPages/EffectUploadPage";
import AccountService from "./services/AccountService";
import EffectService from "./services/EffectService";
import EffectDetailPage from "./pages/EffectPages/EffectDetailPage";

function App() {
  const service = new AccountService();
  const effectService = new EffectService();
  const [stateUser, setStateUser] = useState(null);
  const [effects, setEffects] = useState([]);

  const value = {
    user: stateUser,
    userLogin: loginUser,
    userLogout: logoutUser,
  };
  function loginUser(stateCredentials) {
    setStateUser(service.parseJwt(stateCredentials));
    service.setUser(stateCredentials);
  }
  function logoutUser() {
    setStateUser(null);
    service.logoutUser();
  }

  function ReloadEffects(state) {
    async function GetEffects() {
      setEffects(await effectService.getAllEffects());
    }
    if (state !== false) GetEffects();
  }

  useEffect(() => {
    let user = service.loginUser();
    if (user !== null && user !== "") {
      setStateUser(service.parseJwt(user));
    }
    console.log(effectService.getAllEffects());
    setEffects(effectService.getAllEffects());

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <div data-testid="app-1">
      <userContext.Provider value={value}>
          <Router>
            <NavigationBar />
            <Routes>
              <Route exact path="/" element={<Homepage allEffects={effects} />} />
              <Route
                exact
                path="/sign-up"
                element={<LoginPage />}
              />

              <Route
                exact
                path="/effect/upload"
                element={<EffectUploadPage reloadCallback={ReloadEffects} />}
              />

              <Route
                exact
                path="/effect/details/:id"
                element={<EffectDetailPage />}
              />


            </Routes>
          </Router>
      </userContext.Provider>
    </div>
  );
}

export default App;
