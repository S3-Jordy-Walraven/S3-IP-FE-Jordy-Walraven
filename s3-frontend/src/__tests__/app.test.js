import { render, screen, cleanup, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "../App";

afterEach(cleanup);

test("should render app", () => {
  render(<App />);
  const canvasElement = screen.getByTestId("app-1");
  expect(canvasElement).toBeInTheDocument();
  expect(canvasElement).toHaveTextContent("Home");
});

test("matches snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
