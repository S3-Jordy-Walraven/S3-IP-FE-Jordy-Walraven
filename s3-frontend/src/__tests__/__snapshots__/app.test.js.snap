// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="app-1"
>
  <div>
    <nav
      className="general-dark-bg navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="text-light nav-link nav-item"
          href="/effect/upload"
          onClick={[Function]}
        >
          Upload
        </a>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            className="me-auto navbar-nav"
            onKeyDown={[Function]}
          />
          <div
            className="navbar-nav"
            onKeyDown={[Function]}
          >
            <div
              className="text-light nav-item"
            >
              Je bent uitgelogd
            </div>
            <div
              className="nav-item dropdown"
            >
              <a
                aria-expanded={false}
                className="dropdown-toggle nav-link"
                data-rr-ui-event-key={null}
                href="#"
                id="collasible-nav-dropdown"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                Profile
              </a>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </div>
  <div
    data-testid="homepage-1"
  >
    <div>
      <h1
        style={
          Object {
            "marginTop": "20px",
            "textAlign": "center",
          }
        }
      >
        Signify
      </h1>
    </div>
    <div
      className="container-fluid"
    >
      <div
        className="row justify-content-md-center"
      />
    </div>
  </div>
</div>
`;
